app/viewModels/additionalReference/index/AddAnotherAdditionalReferenceViewModel.scala

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package viewModels.additionalReference.index
+
+import config.FrontendAppConfig
+import controllers.additionalReference.index.routes
+import models.{ArrivalId, Index, Mode, RichOptionalJsArray, UserAnswers}
+import pages.additionalReference.AdditionalReferenceNumberPage
+import pages.sections.additionalReference.AdditionalReferencesSection
+import play.api.i18n.Messages
+import play.api.libs.json.JsArray
+import play.api.mvc.Call
+import viewModels.{AddAnotherViewModel, ListItem}
+
+case class AddAnotherAdditionalReferenceViewModel(listItems: Seq[ListItem], onSubmitCall: Call, nextIndex: Index) extends AddAnotherViewModel {
+  override val prefix: String = "additionalReference.index.addAnotherAdditionalReference"
+
+  override def maxCount(implicit config: FrontendAppConfig): Int = config.maxAdditionalReferences
+
+  override def title(implicit messages: Messages): String = messages(s"$prefix.$emptyOrSingularOrPlural.title", count)
+
+  override def heading(implicit messages: Messages): String = messages(s"$prefix.$emptyOrSingularOrPlural.heading", count)
+
+  override def legend(implicit messages: Messages): String =
+    if (count > 0) messages(s"$prefix.label", count) else messages(s"$prefix.empty.label", count)
+
+  override def maxLimitLabel(implicit messages: Messages): String = messages(s"$prefix.maxLimit.label")
```

Comment:
You don't need to override these as their definition is not different from what we have in the *AddAnotherViewModel* trait we are extending.